rspotify-model/src/idtypes.rs

```diff
@@ -17,51 +88,56 @@ mod private {
     pub trait Sealed {}
 }
 
-pub trait IdType: private::Sealed + Send + Sync {
+pub trait IdType:
+    private::Sealed + Clone + Copy + Debug + PartialEq + Eq + Hash + Serialize + Send + Sync
+{
     const TYPE: Type;
 }
 pub trait PlayableIdType: IdType {}
 pub trait PlayContextIdType: IdType {}
 
-macro_rules! sealed_types {
-    ($($name:ident),+) => {
+/// This macro helps consistently define ID types by implementing the sealed
+/// trait and creating aliases.
+macro_rules! sealed_idtypes {
+    ($($name:ident => $alias:ident, $alias_buf:ident);+) => {
         $(
-            #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+            #[doc = "Please refer to [`crate::idtypes`] for more information."]
+            #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
             pub enum $name {}
             impl private::Sealed for $name {}
             impl IdType for $name {
                 const TYPE: Type = Type::$name;
             }
+
+            #[doc = concat!("Alias for `Id<", stringify!($name), ">`. Please refer to [`crate::idtypes`] for more information.")]
+            pub type $alias = Id<$name>;
+            #[doc = concat!("Owned type for [`", stringify!($alias), "`]. Please refer to [`crate::idtypes`] for more information.")]
+            pub type $alias_buf = IdBuf<$name>;
         )+
     }
 }
 
-sealed_types!(Artist, Album, Track, Playlist, User, Show, Episode);
-
+sealed_idtypes!(
+    Unknown => UnknownId, UnknownIdBuf;
+    Artist => ArtistId, ArtistIdBuf;
+    Album => AlbumId, AlbumIdBuf;
+    Track => TrackId, TrackIdBuf;
+    Playlist => PlaylistId, PlaylistIdBuf;
+    User => UserId, UserIdBuf;
+    Show => ShowId, ShowIdBuf;
+    Episode => EpisodeId, EpisodeIdBuf
+);
+
+impl PlayContextIdType for Unknown {}
 impl PlayContextIdType for Artist {}
 impl PlayContextIdType for Album {}
-impl PlayableIdType for Track {}
 impl PlayContextIdType for Playlist {}
 impl PlayContextIdType for Show {}
+impl PlayableIdType for Unknown {}
+impl PlayableIdType for Track {}
 impl PlayableIdType for Episode {}
 
-pub type ArtistId = Id<Artist>;
-pub type AlbumId = Id<Album>;
-pub type TrackId = Id<Track>;
-pub type PlaylistId = Id<Playlist>;
-pub type UserId = Id<User>;
-pub type ShowId = Id<Show>;
-pub type EpisodeId = Id<Episode>;
-
-pub type ArtistIdBuf = IdBuf<Artist>;
-pub type AlbumIdBuf = IdBuf<Album>;
-pub type TrackIdBuf = IdBuf<Track>;
-pub type PlaylistIdBuf = IdBuf<Playlist>;
-pub type UserIdBuf = IdBuf<User>;
-pub type ShowIdBuf = IdBuf<Show>;
-pub type EpisodeIdBuf = IdBuf<Episode>;
-
-/// A Spotify object id of given [type](crate::enums::types::Type).
+/// A Spotify object ID of a given [type](crate::enums::types::Type).
```

Comment:
I am a little confused that what's difference between the old Id wrapper and the new Id wrapper, it seems the only difference is that alias type declarations are moved into macro `sealed_idtypes` and an `Unknown` type has been added?